### PR TITLE
Queue worker construct

### PR DESCRIPTION
**Pull request by aspcanada**
I've refactored the api.py so that all of the "worker" resources are encapsulated into it's own construct. Please pay attention to the `cdk diff` in the comments after the diff runs.

### 🚨 PLEASE NOTE 🚨
This will tear down the existing SQS Queue so any message in the queue will be lost.

### CODE REVIEW

File: iac/stacks/constructs/worker.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            evaluation_periods=1,
+            comparison_operator=cw.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+            treat_missing_data=cw.TreatMissingData.IGNORE,
+        )
+
+        # CloudWatch Alarm for DLQ
+        dlq_alarm = cw.Alarm(
+            self,
+            "DLQAlarm",
+            metric=queue.metric_approximate_number_of_messages_visible(
```
Comment (saanobhaai) [inline iac/stacks/constructs/worker.py:78]: Shouldn't this be `dead_letter_queue`?

File: iac/stacks/constructs/worker.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                appscaling.ScalingInterval(lower=1, change=+1),
+            ],
+        )
+
+        # allow worker access to public bucket
+        public_bucket.grant_read_write(fargate_service.task_definition.task_role)
+
+        # exports
+        self.queue = queue
+        # self.fargate_service = fargate_service
```
Comment (saanobhaai) [inline iac/stacks/constructs/worker.py:131]: delete comment?

File: iac/stacks/constructs/worker.py (hunk 1 above)
Comment (aspcanada) [reply iac/stacks/constructs/worker.py:80]: yes, thank you!

File: iac/stacks/constructs/worker.py (hunk 2 above)
Comment (aspcanada) [reply iac/stacks/constructs/worker.py:133]: done

**Pull request merged**